StyleImageDownloader/Program.cs

```
@@ -0,0 +1,83 @@
+            }
+
+            return sourceMetadata.Value == value;
+        }
+
+        public static async Task Main(string[] args)
+        {
+            var logger = GetLogger<Program>();
+            Program program = new Program();
+            HttpClient api = new HttpClient();
+            var data_directory = "C:\\BESTSELLER\\style images";
+
+            //get the list of styles
+            var apistyledata = api.GetAsync(program.apiEndpoint).Result;
+            var styleList = JsonConvert.DeserializeObject<List<ApiData>>(apistyledata.ToString());
+
+            foreach (var item in styleList)
+            {
+                var httpclient = new HttpClient();
+
+                Task.Run(async () =>
+                {
+                    try
+                    {
+                        var data = await httpclient.GetStreamAsync(item.ImageUrls[0]);
+
+                        if (MetaDataContainsValue(data, "photoshop"))
+                        {
+                            string folder = data_directory + "\\" + "photoshop" + "\\" + item.StyleName + "_" + item.StyleNumber + ".jpeg";
+
+                            var file = File.Open(folder, FileMode.OpenOrCreate);
+                            data.CopyToAsync(file).Wait();
+                        }
+
+                        if (MetaDataContainsValue(data, "apple-photos"))
+                        {
+                            string folder = data_directory + "\\" + "apple-photos" + "\\" + item.StyleName + "_" + item.StyleNumber + ".jpeg";
+
+                            var file = File.Open(folder, FileMode.OpenOrCreate);
+                            data.CopyToAsync(file).Wait();
+                        }
+                        else
+                        {
+                            logger.LogInformation("Did not know where to save this file");
+                        }
+                    }
+                    catch (Exception ex)
+                    { }
```

`catch` clause is too broad and silently ignores any errors. The clause should at least log errors so that we can know if anything went wrong in the code above